### DISCUSSION TITLE
Why does a user need to register a solver?

### DISCUSSION THREAD

**Discussion by mbakker7** [modflowpy/flopy#2354] [General]
When doing flow and transport, two separate solvers (IMS) need to be specified, one for the gwf model and one for the gwt model. Also with different file names. I can do that, but why do I need to separately register the solvers with the simulation? Something like 
```
sim.register_ims_package(solution_file=gwf_ims, # name of iterative model solver instance
                         model_list=[gwf.name], # list with name of groundwater flow model
)
```
Why is model_list (or just model) not a separate keyword argument when creating the IMS? Something like
```
gwf_ims  = fp.mf6.ModflowIms(simulation=sim, # add to simulation called sim
                             model=gwf, # SPECIFY MODEL HERE
                             filename=gwf.name + '.ims', # file name to store ims
                             linear_acceleration="BICGSTAB", # use BIConjuGantGradientSTABalized method
                            ) 
```
and then there is no need to specify the filename anymore either, just use gwf.name by default. 
And while we are on it, when would you specify a model_list rather than just one model? Any use-case for that?

**Comment by christianlangevin** [comment]
Hey @mbakker7, I don't remember the specifics for why the syntax is the way it is.  I agree that it's cumbersome and not consistent with how we do things.  I think we should consider changing this in our next major version.  There is an option to have multiple "solution groups" in a mf6 simulation and I suspect it may somehow be related to that.

It's true that the common use case for flow and transport is two separate IMS files with one model in each.  But we can also use a single IMS to solve multiple models, such as for LGR applications with parent and child models, or for the parallel case where we solve multiple flow models in a single parallel solution.  We are also working on new surface water models (both channel and overland flow) and these can be solved simultaneously in a single IMS with one or more groundwater flow models.  Also, as you know, we can solve a freshwater model and a saltwater model together in the same numerical solution!

**Reply by mbakker7** [reply]
Thanks for the clarification, @langevin-usgs. I can obviously work with the current flopy implementation, but it is difficult to explain in classes and workshops (other than: this is what you have to do). Especially when you are used to 'just' doing a groundwater model. In that case you simply add an IMS to the simulation and you are done. 
I will submit an issue with a proposed enhancement for possible inclusion in the next major release.

**Reply by mbakker7** [reply]
Opened issue https://github.com/modflowpy/flopy/issues/2355
